napari/layers/labels/labels.py

```diff
@@ -649,6 +666,19 @@ def _raw_to_displayed(self, raw):
         else:
             raise ValueError("Unsupported Color Mode")
 
+        if self.contour:
+            image = np.zeros_like(raw)
+            struct_elem = ndi.generate_binary_structure(raw.ndim, 1)
+            for label in np.unique(raw):
```

Comment:
> @kir0ul but why are you looping over the labels? find_boundaries does it using one dilation, one erosion, and image arithmetic, rather than, in the current case, as many dilations as there are labels...

has this been addressed. i think we'd really want to avoid looping over the labels as that will be very slow. i havn't looked at the underlying mathematics, but feel like it should be possible

Reply:
Not yet, still working on it. I probably missed something but I wasn't able to get the correct boundaries without the loop on the labels. But it has to be possible without the loop since they do it in `skimage`.

Reply:
nice, looks like this has been fixed now!!